src/index.js

```diff
@@ -148,14 +125,26 @@ function getInterface(proto) {
  */
 function getPropertyNames(proto) {
     const results = []
-
     while (proto !== Object.prototype) {
         Object.getOwnPropertyNames(proto).reduce((arr,k) => arr.indexOf(k) === -1 ? arr.push(k) && arr : arr, results)
         proto = Object.getPrototypeOf(proto).constructor.prototype
     }
-
     return results
 }
 
+function isMixin(x) {
+    return (typeof x == 'function') && !!x.mixin
+}
+
+function isClass(x) {
+    return (typeof x == 'function') && (s => /^class\s/.test(s)
+        || /^.*classCallCheck\(/.test(s.replace(/^[^{]*{\s*/,'').replace(/\s*}[^}]*$/,'')))(x.toString())
```

Comment:
> It just saves some bytes.

KISS, make sure it is readable. The rest should be handled by the transpiler. Also `const s =` vs `=>()` is safing two bytes at the expense of performance (the interpreter has to allocate a new closure) and readability. I feel that our internet today is fast enough to handle those two bytes ;)